app/views/tags/index.html.erb

```diff
@@ -26,9 +26,17 @@
 
 <% if @tags %>
   <% if params[:show] == "random" %>
-    <p><%= t(".about.random", search_tags_link: (link_to t(".search_tags"), search_tags_path)).html_safe %></p>
+    <% if @collection %>
+      <p><%= t(".about.random_in_collection", search_tags_link: (link_to t(".search_tags"), search_tags_path)).html_safe %></p>
```

Comment:
For the `@collection` logic here, you can remove the code for the tag search link and the `html_safe` call, since the text for the collection page doesn't include the sentence about tag search. (The issue wasn't very clear about whether it should be included, and I think you made the right call leaving it out.)

Reply:
I have made the requested changes!